subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -347,7 +349,7 @@ static int ieee802154_send(struct net_if *iface, struct net_pkt *pkt)
 		}
 #else
 
-		if (buf->len > IEEE802154_MTU) {
+		if (buf->len > ieee802154_get_msdu_size()) {
```

Comment:
Shouldn't the size be a `struct *_data` area of some device? I guess that is a completely different problem though. Just a comment - I don't think that's something that can be fixed right now. The assumption everywhere in Zephyr is that there is 1 radio, but we need to facilitate multi-instance drivers.

Reply:
Yes, you are right. I forget to pass the `iface` to be checked. I'll fix and review the others functions.